File: app/src/main/java/com/mmutert/freshfreezer/ui/UnitArrayAdapter.java
package com.mmutert.freshfreezer.ui;

import android.content.Context;
import android.widget.ArrayAdapter;

import androidx.annotation.NonNull;

import com.mmutert.freshfreezer.data.AmountUnit;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;


public class UnitArrayAdapter extends ArrayAdapter<CharSequence>{

    private final List<CharSequence> values;

    public UnitArrayAdapter(@NonNull final Context context, final int resource) {
        super(context, resource);

        values = new ArrayList<>();
        for (AmountUnit value : AmountUnit.values()) {
            String valueString = context.getResources().getString(value.getStringResId());
            values.add(valueString);
        }
        super.addAll(values);
    }

    public AmountUnit getSelectedUnit(final int position) {
        return AmountUnit.values()[position];
    }

    public int getIndexOfUnit(AmountUnit unit) {
        return Arrays.asList(AmountUnit.values()).indexOf(unit);
    }
}

File: app/src/main/java/com/mmutert/freshfreezer/ui/UnitArrayAdapter.kt
package com.mmutert.freshfreezer.ui

import android.content.Context
import android.widget.ArrayAdapter
import com.mmutert.freshfreezer.data.AmountUnit

class UnitArrayAdapter(context: Context, resource: Int) :
    ArrayAdapter<CharSequence?>(context, resource) {

    private val values: MutableList<CharSequence> = ArrayList()

    fun getSelectedUnit(position: Int): AmountUnit {
        return AmountUnit.values()[position]
    }

    fun getIndexOfUnit(unit: AmountUnit): Int {
        return AmountUnit.values().indexOf(unit)
    }

    init {
        for (value in AmountUnit.values()) {
            val valueString = context.resources.getString(value.stringResId)
            values.add(valueString)
        }
        super.addAll(values)
    }
}